packages/api/src/command/medical/document/document-reconvert.ts

```diff
@@ -0,0 +1,318 @@
+import { DocumentReference } from "@medplum/fhirtypes";
+import { resourceTypeForConsolidation } from "@metriport/api-sdk";
+import { S3Utils } from "@metriport/core/external/aws/s3";
+import { executeAsynchronously } from "@metriport/core/util/concurrency";
+import { out } from "@metriport/core/util/log";
+import { capture } from "@metriport/core/util/notifications";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { groupBy } from "lodash";
+import { DocRefMapping } from "../../../domain/medical/docref-mapping";
+import { convertCDAToFHIR } from "../../../external/fhir-converter/converter";
+import { getDocuments as getDocumentsFromFHIRServer } from "../../../external/fhir/document/get-documents";
+import { downloadedFromHIEs } from "../../../external/fhir/shared";
+import { getMetriportContent } from "../../../external/fhir/shared/extensions/metriport";
+import { PatientModel as Patient } from "../../../models/medical/patient";
+import { Config } from "../../../shared/config";
+import { errorToString } from "../../../shared/log";
+import { formatNumber } from "../../../shared/numbers";
+import { getDocRefMappings } from "../docref-mapping/get-docref-mapping";
+import { deleteConsolidated as deleteConsolidatedOnFHIRServer } from "../patient/consolidated-delete";
+import { getConsolidated } from "../patient/consolidated-get";
+import { getPatientOrFail } from "../patient/get-patient";
+import { makeDocRefContentToFileFunction, SimplerFile } from "./document-query-storage-info";
+
+dayjs.extend(duration);
+
+// TODO review this
+const patientsToProcessInParallel = 1;
+
+const resourcesToDelete = resourceTypeForConsolidation.filter(r => r !== "DocumentReference");
+const s3Utils = new S3Utils(Config.getAWSRegion());
+
+export type ReConvertDocumentsCommand = {
+  cxId: string;
+  patientIds?: string[];
+  documentIds?: string[];
+  dateFrom?: string;
+  dateTo?: string;
+  requestId: string;
+  logConsolidatedCountBeforeAndAfter?: boolean;
+};
+
+type DocumentReferenceWithId = DocumentReference & { id: string };
+
+type DocRefWithS3Info = {
+  docRef: DocumentReferenceWithId;
+  file: SimplerFile;
+};
+
+export const reConvertDocuments = async (params: ReConvertDocumentsCommand): Promise<void> => {
+  const startedAt = Date.now();
+  const {
+    cxId,
+    patientIds = [],
+    documentIds = [],
+    dateFrom,
+    dateTo,
+    requestId,
+    logConsolidatedCountBeforeAndAfter = false,
+  } = params;
+  const { log } = out(`reConvertDocuments - cxId ${cxId}`);
+  log(
+    `Re-converting documents: ${documentIds.length ?? "all"}, ` +
+      `patients: ${patientIds.length ?? "all"}, ` +
+      `dateFrom: ${dateFrom}, dateTo: ${dateTo} - ` +
+      `docs: ${documentIds.join(", ")}; ` +
+      `patients: ${patientIds.join(", ")}`
```

Comment:
nit: `patientIds` would be more descriptive - wasn't sure why we had `patients` twice at first glance

Reply:
I think its NABD, when we see the log it's very clear one is count and the other is the list of ids